### DISCUSSION TITLE
How to create a DataArray from a dask.array?

### DISCUSSION THREAD

**Discussion by ianliu** [pydata/xarray#6268] [Q&A] [answered]
Is it possible to create a `DataArray` from a `dask.array`? The API documentation for `xarray.DataArray` says the following for the _data_ argument:

> **data** ([array_like](https://numpy.org/doc/stable/glossary.html#term-array_like)) – Values for this array. Must be an numpy.ndarray, ndarray like, or castable to an ndarray. If a self-described xarray or pandas object, attempts are made to use this array’s metadata to fill in other unspecified arguments. A view of the array’s data is used instead of a copy if possible.

It doesn't mention anything about passing a _dask.array_, but the following snippet of code seems to work as expected:

```python
import numpy as np
import dask
import dask.array as da
import xarray as xr

arr = da.stack([
    da.from_delayed(dask.delayed(np.ones)(5),
                    shape=(5,),
                    dtype=float)
    for _ in range(3)])

xr.DataArray(arr, coords={"y": list("abc"), "x": list(range(5))})
```

Can I assume that the aforementioned snippet of code doesn't compute the dask array to create the _DataArray_?

**Comment by keewis** [comment, marked as answer]
yes, you can rely on `dask` not to compute if you pass `dask` arrays to the constructor of `DataArray`.

I guess this part of the docstring has not been updated since quite a while? It is supposed to refer to [`duck array`](https://docs.xarray.dev/en/stable/user-guide/terminology.html#term-duck-array) instead of `array_like`, since that's what we actually support.
